src/morale.cpp

```diff
@@ -357,66 +357,85 @@ void player_morale::decay( int ticks )
 
 void player_morale::display( double focus_gain )
 {
-    // Create and draw the window itself.
-    WINDOW *w = newwin( FULL_SCREEN_HEIGHT, FULL_SCREEN_WIDTH,
-                        ( TERMY > FULL_SCREEN_HEIGHT ) ? ( TERMY - FULL_SCREEN_HEIGHT ) / 2 : 0,
-                        ( TERMX > FULL_SCREEN_WIDTH ) ? ( TERMX - FULL_SCREEN_WIDTH ) / 2 : 0 );
-    draw_border( w );
+    const char *morale_gain_caption = _( "Total morale gain" );
+    const char *focus_gain_caption = _( "Focus gain per minute" );
 
-    // Figure out how wide the name column needs to be.
-    int name_column_width = 18;
+    // Figure out how wide the source column needs to be.
+    int source_column_width = utf8_width( focus_gain_caption );
+    if( utf8_width( morale_gain_caption ) > source_column_width ) {
+        source_column_width = utf8_width( morale_gain_caption );
```

Comment:
Instead of repeating the function call, you can use `std::max` here: `width = std::max(width, utf8_width(...));`

Same below.